controllers/fetchdata_controller.go

```diff
@@ -74,17 +77,17 @@ func (r *FetchDataReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
-	if err := r.createCronJob(FetchData); err != nil {
+	if err := r.createCronJob(FetchData, r.DB.Config()); err != nil {
 		l.Error(err, "failed to create the CronJob resource")
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *FetchDataReconciler) createCronJob(m *curatorv1alpha1.FetchData) error {
+func (r *FetchDataReconciler) createCronJob(m *curatorv1alpha1.FetchData, dbConfig *pgx.ConnConfig) error {
```

Comment:
Same here: we can now drop the dbConfig parameter. Let's also update the function signature to take in a context.Context parameter:

```go
func (r *FetchDataReconciler) createCronJob(ctx context.Context, m *curatorv1alpha1.FetchData) error {
    ...
}
```